frames/src/test/java/com/checkout/frames/component/expirydate/ExpiryDateViewModelTest.kt

```diff
@@ -0,0 +1,287 @@
+        val smartExpiryDateValidationRequest = SmartExpiryDateValidationRequest(true, testExpiryDate)
+
+        every {
+            mockSmartExpiryDateValidationUseCase.execute(eq(smartExpiryDateValidationRequest))
+        } returns ValidationResult.Success(testExpiryDate)
+        // Show error from the start to check that it will be hidden.
+        viewModel.componentState.showError(R.string.cko_base_invalid_past_expiry_date_error)
+
+        // When
+        viewModel.onExpiryDateInputChange(testExpiryDate)
+
+        // Then
+        verify(exactly = 1) { mockSmartExpiryDateValidationUseCase.execute(eq(smartExpiryDateValidationRequest)) }
+        with(viewModel.componentState.inputState) {
+            Assertions.assertFalse(inputFieldState.isError.value)
+            Assertions.assertFalse(errorState.isVisible.value)
+        }
+        Assertions.assertEquals(viewModel.componentState.expiryDate.value, testExpiryDate)
+        Assertions.assertEquals(viewModel.componentState.expiryDateMaxLength.value, EXPIRY_DATE_MAXIMUM_LENGTH_THREE)
+    }
+
+    @ParameterizedTest(
+        name = "When on expiry date change invoked with {0} then {1} validation (maxLength = {2}) set to field state"
+    )
+    @MethodSource("onTextChangedArguments")
+    fun `When expiryDate change called with valid input then expected text & maxLength should be set to componentState`(
+        enteredText: String,
+        expectedText: String,
+        maxLength: Int
+    ) {
+        // Given
+        val smartExpiryDateValidationRequest = SmartExpiryDateValidationRequest(true, enteredText)
+        every {
+            mockSmartExpiryDateValidationUseCase.execute(eq(smartExpiryDateValidationRequest))
+        } returns ValidationResult.Success(enteredText)
+
+        // When
+        viewModel.onExpiryDateInputChange(enteredText)
+
+        // Then
+        Assertions.assertEquals(viewModel.componentState.expiryDateMaxLength.value, maxLength)
+        Assertions.assertEquals(viewModel.componentState.expiryDate.value, expectedText)
+    }
+
+    companion object {
+        @JvmStatic
+        fun onTextChangedArguments(): Stream<Arguments> = Stream.of(
+            Arguments.of("123", "123", EXPIRY_DATE_MAXIMUM_LENGTH_FOUR),
+            Arguments.of("234", "234", EXPIRY_DATE_MAXIMUM_LENGTH_THREE),
+            Arguments.of("1254", "1254", EXPIRY_DATE_MAXIMUM_LENGTH_FOUR),
+            Arguments.of("1234", "1234", EXPIRY_DATE_MAXIMUM_LENGTH_FOUR),
+            Arguments.of("116", "116", EXPIRY_DATE_MAXIMUM_LENGTH_FOUR),
+            Arguments.of("223", "223", EXPIRY_DATE_MAXIMUM_LENGTH_THREE),
+            Arguments.of("223", "223", EXPIRY_DATE_MAXIMUM_LENGTH_THREE),
+            Arguments.of("445", "445", EXPIRY_DATE_MAXIMUM_LENGTH_THREE),
+            Arguments.of("333", "333", EXPIRY_DATE_MAXIMUM_LENGTH_THREE),
+            Arguments.of("1187", "1187", EXPIRY_DATE_MAXIMUM_LENGTH_FOUR)
+        )
+    }
```

Comment:
It is verifying the entered text along with the max length and expected text (which should be exactly the same for valid input). Do you want to remove this expected text if eventually it is the same and leave it as it is the max length checks by user input?

Reply:
Yes, that's what I thought about.
In the current implementation, text change is not a responsibility of the view model, so entered and expected data will be the same.
Except for cases when non-digits will be pathed, but it will be covered by another test.